### PR TITLE
Backport v1.5 2019 05 07

### PR DESCRIPTION
**Pull request by raybejjani**
* PR: 7887 -- Annotate node status with NetworkUnavailable as false (@aanm) -- https://github.com/cilium/cilium/pull/7887                                                                                                                                                                                                                     
* PR: 7922 -- pkg/metrics: add Cilium namespace to fqdn_gc_deletions_total (@aanm) -- https://github.com/cilium/cilium/pull/7922
 * PR: 7906 -- Patch, instead of Update, node annotations (@aanm) -- https://github.com/cilium/cilium/pull/7906
 * PR: 7866 -- bpf: Set BPF_F_NO_PREALLOC before comparing maps (@brb) -- https://github.com/cilium/cilium/pull/7866
 * PR: 7927 -- pkg/buildqueue: remove unused package (@aanm) -- https://github.com/cilium/cilium/pull/7927
 * PR: 7896 -- add option to enable or disable from default metrics (@aanm) -- https://github.com/cilium/cilium/pull/7896
 * PR: 7919 -- test: Do not set enable-legacy-services in v1.4 ConfigMap (@brb) -- https://github.com/cilium/cilium/pull/7919
 * PR: 7931 -- docs,examples: Fix up custom CNI for microk8s (@joestringer) -- https://github.com/cilium/cilium/pull/7931

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7940)
<!-- Reviewable:end -->

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-missed-k8s

**Comment by raybejjani** [comment]
test-docs-please

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-missed-k8s

**Comment by raybejjani** [comment]
test-docs-please

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -535,111 +343,645 @@ var (
 	//
 	// Deprecated: This metric can be removed in 1.6 as
 	// KVStoreOperationsDuration provides the count along with duration
-	KVStoreOperationsTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
```
Comment (raybejjani) [inline pkg/metrics/metrics.go:538]: @aanm I had to convert this metric, which doesn't exist in 1.6.

Comment (aanm) [reply pkg/metrics/metrics.go:538]: Looks good

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-missed-k8s

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-missed-k8s

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-missed-k8s

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-missed-k8s

**Pull request merged**